src/customizedhooks/useAxios.js

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import {useState, useEffect} from "react";
+
+const axiosClient = axios.create({
+    baseURL: "https://api.coingecko.com/api/v3"
+})
+
+const useAxios = (url)=>{
+    
+    const [data, setData] = useState()
+    const [status,setStatus] = useState("")
+    const [loader, setLoader] = useState(true)
+
+    useEffect(() => {
+
+        (async function (){
+            try{
+                const response = await axiosClient.get(url)
+                setStatus(response.status)
+                setData(response.data);
+            }catch(error){
+                setStatus(error)
+            }finally{
+                setLoader(false)
+            }
```

Comment:
Won't it make loader always false?

Reply:
because either we got error or not we should finish the Loading process, it's an asynchrone call it will take some time so when we got the error or the data after that we should finish the Loading process.

Reply:
an for each call we will init the Loader to True and after receiving any response from the Server we will change it to false cuz there no Loading process since we got already the response.